src/disco/containers/InstallButton.js

```diff
@@ -0,0 +1,63 @@
+import React, { PropTypes } from 'react';
+import { connect } from 'react-redux';
+
+import { gettext as _ } from 'core/utils';
+
+import 'disco/css/InstallButton.scss';
+import {
+  DOWNLOADING,
+  INSTALLED,
+  UNKNOWN,
+  validInstallStates as validStates,
+} from 'disco/constants';
+
+export class InstallButton extends React.Component {
+  static propTypes = {
+    handleClick: PropTypes.func,
+    handleChange: PropTypes.func,
+    guid: PropTypes.string,
+    progress: PropTypes.number,
+    slug: PropTypes.string.isRequired,
+    state: PropTypes.oneOf(validStates),
+  }
+
+  static defaultProps = {
+    state: UNKNOWN,
```

Comment:
I feel this is a bit confusing since there can be state in a react component. That's why I used addonState before so that it clears up the distinction because there could be `this.state` but this is `this.props.state`.